src/state/debugger.rs

```diff
@@ -47,17 +59,22 @@ impl Debugger {
         let contract = contract.copied().unwrap_or_default();
         let last_state = self.last_state.take();
 
+        let current = Breakpoint::raw(contract, pc);
+
+        if self.single_stepping {
+            return match last_state {
+                Some(s) if s == current => DebugEval::Continue,
+                _ => current.into(),
+            };
+        }
+
         self.breakpoints
             .get(&contract)
             .map(|set| set.get(&pc))
             .flatten()
-            .map(|_| {
-                let breakpoint = Breakpoint::raw(contract, pc);
-
-                match last_state {
-                    Some(s) if s == breakpoint => DebugEval::Continue,
-                    _ => breakpoint.into(),
-                }
+            .map(|_| match last_state {
+                Some(s) if s == current => DebugEval::Continue,
+                _ => current.into(),
```

Comment:
What's the point of these breakpoints if we just ignore their locations `map(|_| ..` here? Shouldn't we have something like this?
```suggestion
            .map(|location| match last_state {
                let breakpoint = Breakpoint::raw(contract, location);
                Some(s) if s == breakpoint => DebugEval::Continue,
                _ => breakpoint.into(),
```

Reply:
Not going to hold up the PR over this since it was already this way before.

Reply:
I'll make a separate PR to address this